File: BackendGL/src/models/employee.go
package models

import (
	"time"
)

type Employee struct {
	ID              uint       `gorm:"primaryKey" json:"id"`
	Name            string     `gorm:"not null" json:"name"`
	Surname         string     `json:"surname"`
	Email           string     `gorm:"unique;not null" json:"email"`
	Password        string     `json:"password"`
	Role            string     `json:"role"`
	StartOfContract time.Time  `json:"start_of_contract"`
	EndOfContract   *time.Time `json:"end_of_contract"`
	Telephone       string     `json:"telephone"`
	Address         string     `json:"address"`
	Token           string     `json:"token"`
	IdEnterprise    uint       `gorm:"column:id_enterprise; not null" json:"enterprise_id"`
	Enterprise      Enterprise `gorm:"foreignKey:IdEnterprise;references:ID" json:"enterprise"`
}

type EmployeeImage struct {
	ID         uint     `gorm:"primaryKey" json:"id"`
	IdEmployee uint     `gorm:"not null" json:"id_employee"`
	Employee   Employee `gorm:"foreignKey:IdEmployee; references:ID;" json:"employee"`
	Image      []byte   `gorm:"type:btyea" json:"image"`
}

func (Employee) TableName() string {
	return "employee"
}

// =================== SALARY ===================

type Salary struct {
	ID            uint      `gorm:"primaryKey" json:"id"`
	EmployeeID    uint      `gorm:"not null" json:"employee_id"`
	Employee      Employee  `gorm:"foreignKey:EmployeeID;constraint:OnUpdate:CASCADE;" json:"employee"`
	Amount        float64   `gorm:"type:numeric(10,2);not null" json:"amount"`
	EffectiveDate time.Time `gorm:"autoCreateTime" json:"effective_date"`
	UpdatedAt     time.Time `gorm:"autoUpdateTime" json:"updated_at"`
}

func (Salary) TableName() string {
	return "salary"
}

File: Backend/src/repositories/user/employee.ts
import { DataTypes, Model } from 'sequelize'
import { Enterprise } from '../enterprise'
import sequelize from '..'


export class Employee extends Model{
  declare id: number
  declare email: string
  declare password?: string
  declare name?: string
  declare surname?: string
  declare enterprise_id?: number
  declare start_of_contract?: string
  declare end_of_contract?: string
  declare role?: string
  declare telephone?: string
  declare address?: string
  declare token?: string
}

Employee.init(
  {
    id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
      },
    email: {
        type: DataTypes.STRING,
        allowNull: false,
        unique: true
    },
    password: {
        type: DataTypes.TEXT,
        allowNull: true,
    },
    name: {
        type: DataTypes.STRING,
        allowNull: true,
    },
    surname: {
        type: DataTypes.STRING,
        allowNull: false,
    },
    enterprise_id: {
        type: DataTypes.INTEGER,
        allowNull: true,
        references: {
            model: Enterprise,
            key: 'id',
        },
    },
    start_of_contract: {
      type: DataTypes.DATE,
      allowNull: true,
    },
    end_of_contract: {
      type: DataTypes.DATE,
      allowNull: true,
    },
    role: {
        type: DataTypes.STRING,
        allowNull: false,
    },
    telephone: {
      type: DataTypes.STRING,
      allowNull: true,
    },
    address: {
      type: DataTypes.STRING,
      allowNull: true,
    },
    token: {
      type: DataTypes.STRING,
      allowNull: true,
    }
  },
  {
    sequelize,
    modelName: 'Employee',
    tableName: 'employee',
    timestamps: false 
  }
)

Employee.belongsTo(Enterprise, {foreignKey: 'enterprise_id', as: 'enterprise'})
Enterprise.hasMany(Employee, { foreignKey: 'enterprise_id', as: 'employees' });